src/ObjectList.php

```diff
@@ -195,6 +196,19 @@ public function exclude(ObjectList $otherList)
     }
 
     /**
+     * Return a new list containing only those elements present in {$otherList}.
+     *
+     * @param ObjectList $otherList
+     * @return static
+     */
+    public function retain(ObjectList $otherList)
```

Comment:
Ok, I checked lodash, underscore, scala and Haskell. They all use intersect, not retain :). But I can accept both, just I know people with more functional world will probably look for intersect not retain.